tests/network/vmi_istio.go

```diff
@@ -0,0 +1,313 @@
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			// Enable sidecar injection by setting the namespace label
+			libnet.AddLabelToNamespace(tests.NamespaceTestDefault, tests.IstioInjectNamespaceLabel, "enabled")
+			defer libnet.RemoveLabelFromNamespace(tests.NamespaceTestDefault, tests.IstioInjectNamespaceLabel)
+
+			ports := []v1.Port{
+				{Port: svcDeclaredTestPort},
+				{Port: svcUndeclaredTestPort},
+			}
+			vmi = newVMIWithIstioSidecar(ports)
+			service = newService()
+
+			By("Create NetworkAttachmentDefinition")
+			nad = &k8snetworkplumbingwgv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "istio-cni",
+				},
+				Spec: k8snetworkplumbingwgv1.NetworkAttachmentDefinitionSpec{},
+			}
+			nad, err = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(tests.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
+			table.DescribeTable("request to VMI", func(port int) {
+				job, err = createJobCheckingVMIReachability(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				peerAuthenticationRes := schema.GroupVersionResource{Group: "security.istio.io", Version: "v1beta1", Resource: "peerauthentications"}
+				var peerAuthentication *unstructured.Unstructured
+				BeforeEach(func() {
+					peerAuthentication = &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"apiVersion": "security.istio.io/v1beta1",
+							"kind":       "PeerAuthentication",
+							"metadata": map[string]interface{}{
+								"name": "strict-pa",
+							},
+							"spec": map[string]interface{}{
+								"mtls": map[string]interface{}{
+									"mode": "STRICT",
+								},
+							},
+						},
+					}
```

Comment:
It's a bit distracting to see this here. Could you consider having it in an helper ? It could even be a lambda in this context - if this is not reused in other scopes. `generatePeerAuthRequest` or something alike ?

Reply:
Done